src/seq.rs

```diff
@@ -601,23 +654,41 @@ impl fmt::Debug for Event {
     }
 }
 
-/// Low level methods to set/get data on an Event. Don't use these directly, use generic methods on Event instead.
-pub trait EventData {
+/// Low level method to check if event type supports a given data type. Don't use directly, use generic methods on Event instead.
+pub trait HasEventData {
```

Comment:
Also, my solution did not introduce a way to retrieve the referenced buffer without copying (i.e., a version of `get_data` with `Event<'a> -> &'a [u8]`. I think, there should be a special cased version for that as well (`get_data_borrowed`, to keep the names consistent?).